src/lib/ext-activitylog.js

```diff
@@ -135,9 +175,31 @@ class View {
     this.keywordFilter = document.querySelector('filter-keyword');
     this.timestampFilter = document.querySelector('filter-timestamp');
 
+    this.optionsBtn.addEventListener('click', this);
     this.clearLogBtn.addEventListener('click', this);
     this.saveLogBtn.addEventListener('click', this);
     this.loadLogFile.addEventListener('change', this);
+    this.logView.addEventListener('logcountchange', this);
+
+    this.extFilter.addEventListener('triggerdropdown', this);
```

Comment:
Every custom element with "dropdown" functionality should manage the dropdown logic itself. This logic is an implementation detail of the custom component, and therefore you should not put listeners in the view here. The components themselves should invoke the DropDownController where needed.

The DropDownController is a singleton; since the components need a reference to an instance, the DropDownController can either export an instance, or the singleton design pattern can be used (`getInstance`).

Reply:
Thanks for mentioning this. I have made it a singleton design. Let me know if I did it correctly.